### PR TITLE
shuffle-benchmark: add `--partition-distribution`

### PR DESCRIPTION
**Pull request by madsbk**
Implements a `--partition-distribution` argument to `local_cudf_shuffle.py`

### CODE REVIEW

**Comment by ajschmidt8** [comment]
I canceled the latest GH Action run because it looks like the python tests were in-progress for over 4 hours, which is dramatically longer than the usual ~30 minutes that they take.

The logs are below, it looks like there were some errors throughout the tests.

- https://github.com/rapidsai/dask-cuda/actions/runs/3912268189/jobs/6686787572


To prevent this from happening again, I will open a PR to put a reasonable timeout on the Python tests.

**Review by wence- — COMMENTED** [review]
Minor niggle in an error message, but otherwise I think this looks fine.

File: dask_cuda/benchmarks/local_cudf_shuffle.py [hunk 1]
```diff
@@ -70,20 +70,37 @@ def create_data(
...
+
+    if len(dist) != len(workers):
+        raise ValueError(
+            f"The length of `--devs`({len(dist)}) and "
+            f"`--partition-distribution`({len(workers)}) doesn't match"
         )
+    if sum(dist) != args.in_parts:
+        raise ValueError(
+            f"The sum of `--partition-distribution`({sum(dist)}) "
+            f"must match the length of `--devs`({len(workers)})"
```
[the remark this answers is not in the document]
Comment (madsbk) [reply dask_cuda/benchmarks/local_cudf_shuffle.py]: Fixed

**Review by wence- — APPROVED** [review]
Thanks!

**Pull request merged**